File: Trees/Traversals/Traversal.py
class Tree:
    def __init__(self,data):
        self.data=data
        self.left=None;
        self.right=None;

    def insert(self,data):
        self.newNode=Tree(data)
        return self.newNode

    def preorder(self,root):
        if(root==None):
           return        
        print("Data is ",root.data)
        self.preorder(root.left)
        self.preorder(root.right)

    def levelOrder(self,root):
        if(root==None):
            return

        #get a queue(list)
        que=[]
        que.append(root)

        while len(que)>0:
            node=que.pop(0)

            #print the data
            print(node.data)  

            #Add Child Node
            if(node.left!=None):
                que.append(node.left)
            if(node.right!=None):        
                que.append(node.right)
                
root=Tree(10)  
root.left=Tree(20)
root.right=Tree(30)
root.left.left=Tree(40)
root.left.right=Tree(50)
root.right.left=Tree(60)
root.right.right=Tree(70)

root.preorder(root)    
root.levelOrder(root)

File: Trees/Traversals/Traversal.cpp
#include<iostream>
#include<queue>
using namespace std;

class Tree{
   int data; 
   
   public: 
   Tree* left;
   Tree* right;
   Tree(){

   }
   Tree(int data){
       this->data=data;
       this->left=NULL;
       this->right=NULL;
   }

   Tree* insertNode(int data){
       Tree* newNode=new Tree(data);
       return newNode;
   }

   void preOrder(Tree *root){
       if(root==NULL)
         return;
       cout<<root->data<<" ";
       preOrder(root->left);
       preOrder(root->right);  
   }

    void inOrder(Tree *root){
       if(root==NULL)
         return;
       inOrder(root->left);
       cout<<root->data<<" ";
       inOrder(root->right);  
   }

    void postOrder(Tree *root){
       if(root==NULL)
         return;
       
       postOrder(root->left);
       postOrder(root->right);  
       cout<<root->data<<" ";
   }

   void levelOrder(Tree* root){
     if(root==NULL)
      return;
     queue<Tree*> qu;

     qu.push(root);
     while(!qu.empty()){
       Tree* node=qu.front();
       qu.pop();

       cout<<node->data<<" ";
       
       if(node->left!=NULL){
         qu.push(node->left);
       }
       if(node->right!=NULL){
         qu.push(node->right);
       }

     }


   }
};

int main(){
    Tree *root=NULL;
    Tree* b=new Tree();
    root=b->insertNode(10);
    root->left=b->insertNode(20);
    root->right=b->insertNode(30);
    root->left->left=b->insertNode(40);
    root->left->right=b->insertNode(50);
    root->right->left=b->insertNode(60);
    root->right->right=b->insertNode(70);

    cout<<"PreOrder Traversal Is\n";
    b->preOrder(root);
    cout<<"\n";

    cout<<"InOrder Traversal Is\n";
    b->inOrder(root);
    cout<<"\n";

    cout<<"PostOrder Traversal Is\n";
    b->postOrder(root);
    cout<<"\n";

    cout<<"LevelOrder Traversal Is\n";
    b->levelOrder(root);
    cout<<"\n";


    return 0;
}